src/common/editor/index.js

```diff
@@ -1,23 +1,44 @@
-import EditorNew, { useMonaco } from '@monaco-editor/react';
+import MonacoEditor, { useMonaco } from '@monaco-editor/react';
 import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
 
-export default function Editor({ code, language, height, onChange }) {
+export default function Editor({ code, language, height, warnings, onChange }) {
 	const monaco = useMonaco();
 
+	const lang = normalizeLanguage(language);
+	const path = `editor.${lang}`;
+
 	useEffect(() => {
-		if (monaco) {
-			monaco.languages.css.cssDefaults.setOptions({ validate: false });
-		}
-	}, [monaco]);
+		if (!monaco) return;
+
+		if (!warnings) return;
+
+		const { Warning, Error } = monaco.MarkerSeverity;
+		const markers = warnings.map(({ text, line, column, endLine, endColumn, severity }) => ({
+			message: text,
+			severity: severity === 'warning' ? Warning : Error,
+			startLineNumber: line,
+			startColumn: column,
+			endLineNumber: endLine,
+			endColumn,
+		}));
+
+		const model = monaco.editor.getModel(monaco.Uri.parse(path));
+
+		monaco.editor.setModelMarkers(model, 'stylelint', markers);
+	}, [monaco, lang, path, warnings]);
 
 	return (
-		<EditorNew
-			language={normalizeLanguage(language)}
+		<MonacoEditor
+			language={lang}
+			path={path}
```

Comment:
[note] See the [`monaco-react` doc](https://github.com/suren-atoyan/monaco-react#multi-model-editor) about the `path` prop.